starter_code/src/movies.js

```diff
@@ -1,20 +1,118 @@
 /* eslint no-restricted-globals: 'off' */
 // Turn duration of the movies from hours to minutes 
 
+function turnHoursToMinutes (array){
+  var mov = array.map(function(elem){
+    var newElem = Object.assign({},elem);
+
+    var time = 0;
+    var dur = newElem.duration.split(' '); 
+    if (dur.length === 1){
+        if (dur[0].indexOf('h') === -1){
+          var min = parseInt(dur[0].slice(0,3));
+          time += min;
+      } else {
+          var hours = parseInt(dur[0].slice(0,1));
+          time += hours*60;      
+      }
+    }
+    else {
+      var hours = parseInt(dur[0].slice(0,1));
+      var min = parseInt(dur[1].slice(0,3));
+      time += hours * 60;
+      time += min;
+      } 
+  newElem.duration = time;
+  return newElem; 
+  }); 
+return mov;
+};
 
 // Get the average of all rates with 2 decimals 
 
+function ratesAverage(array){
+  var rates = array.reduce(function(acc,film){
+    acc += parseFloat(film.rate);
+    return acc;
+  },0);
+  var average = rates / array.length;
+  return Math.round(average * 100)/100;
+};
+
 
 // Get the average of Drama Movies
 
+function dramaMoviesRate(array){
+  var dramaMovies = 0;
+  var dramaRates = array.reduce(function(acc,film){
+    for (var i = 0; i < film.genre.length; i++){
+        if (film.genre[i]==='Drama'){
+        acc += parseFloat(film.rate);
+        dramaMovies++;
+        return acc;
```

Comment:
El segundo test fallido, `Only Drama Movies! You should return the average of Drama movies only!`, te da porque devuelves la variable acumuladora dentro del `if`. Esto produce que rompa la ejecución del bucle `for` y devuelva la variable acumuladora solo dentro del bloque `for`. Pero no lo devuelve desde el bloque de la función `reduce`. Este problema se resuelve si sacas el `return acc` del `if` y del `for` haces que lo devuelva la función `filter`